spirv_glsl.cpp

```diff
@@ -9778,6 +9778,20 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
 	case OpNoLine:
 		break;
 
+	case OpDemoteToHelperInvocationEXT:
+		if (!options.vulkan_semantics)
+			SPIRV_CROSS_THROW("GL_EXT_demote_to_helper_invocation is only supported in Vulkan GLSL.");
+		require_extension_internal("GL_EXT_demote_to_helper_invocation");
+		statement(backend.demote_literal, ";");
+		break;
+
+	case OpIsHelperInvocationEXT:
+		if (!options.vulkan_semantics)
+			SPIRV_CROSS_THROW("GL_EXT_demote_to_helper_invocation is only supported in Vulkan GLSL.");
+		require_extension_internal("GL_EXT_demote_to_helper_invocation");
+		emit_op(ops[0], ops[1], "helperInvocationEXT()", true);
```

Comment:
This expression cannot be forwarded. If this is forwarded, and we then call Demote(), and then consume the forwarded expression, we'll query IsHelperInvocation after the fact. Alternatively, DemoteToHelperInvocation() must invalidate all IsHelperInvocationEXT expressions, but that's a bit overkill.

Reply:
There should be a test which exposes this.